ttnn/cpp/ttnn/operations/ccl/all_gather/device/multi_core/all_gather_op_multi_core.cpp

```diff
@@ -688,7 +690,7 @@ operation::ProgramWithCallbacks all_gather_multi_core_with_workers_helper(
                 }
                 if (rem_pages != 0) {
                     rem_pages_per_worker.at(worker_idx % all_gather_config.get_num_workers_per_link()) = rem_pages;
-                    TT_ASSERT(rem_pages_per_worker.at(worker_idx % all_gather_config.get_num_workers_per_link()) <= cb_num_pages);
+                    TT_ASSERT(rem_pages_per_worker.at(worker_idx % all_gather_config.get_num_workers_per_link()) <= max_pages_per_chunk * 2);
```

Comment:
nit: I think now that you've pulled out the `div 2` from the max pages per chunk that's passed to kernels, you may as well also compute this value up top with a name: "cb_size_pages", so that way we can have flexibility to change chunks per CB. Not fundamental issue.